plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/src/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.xtend

```diff
@@ -50,56 +56,107 @@ class DslProposalProvider extends AbstractDslProposalProvider {
 	}
-	
-	override completeFunction_Name(EObject model, Assignment assignment, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
+
+	override completeFunction_Name(EObject model, Assignment assignment, ContentAssistContext context,
+		ICompletionProposalAcceptor acceptor) {
 		getExistingParticipants("function", context, acceptor)
 	}
-	
-	override completeActivity_Name(EObject model, Assignment assignment, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
+
+	override completeActivity_Name(EObject model, Assignment assignment, ContentAssistContext context,
+		ICompletionProposalAcceptor acceptor) {
 		getExistingParticipants("activity", context, acceptor)
 	}
-	
-	override completeEntity_Name(EObject model, Assignment assignment, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
+
+	override completeEntity_Name(EObject model, Assignment assignment, ContentAssistContext context,
+		ICompletionProposalAcceptor acceptor) {
 		getExistingParticipants("entity", context, acceptor)
 	}
-	
-	override completeRole_Name(EObject model, Assignment assignment, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
+
+	override completeRole_Name(EObject model, Assignment assignment, ContentAssistContext context,
+		ICompletionProposalAcceptor acceptor) {
 		getExistingParticipants("role", context, acceptor)
 	}
-	
+
 	def getExistingParticipants(String keyword, ContentAssistContext context, ICompletionProposalAcceptor acceptor) {
 		for (String el : EmbeddedEditorInstanceHelper.getAvailablePartNames(keyword)) {
 			acceptor.accept(createCompletionProposal("\"" + el + "\"", "\"" + el + "\"", null, context));
-		} 
+		}
 	}
 
 	override completeSequenceMessage_Source(EObject model, Assignment assignment, ContentAssistContext context,
 		ICompletionProposalAcceptor acceptor) {
 		for (EObject el : variablesDefinedBefore2(model as Model)) {
-			acceptor.accept(createCompletionProposal((el as Actor).name, (el as Actor).name, null, context))
+			createAcceptorForParticipants(el, acceptor, context)
 		}
 	}
 
 	override completeSequenceMessage_Target(EObject model, Assignment assignment, ContentAssistContext context,
 		ICompletionProposalAcceptor acceptor) {
 		for (EObject el : variablesDefinedBefore3(model as SequenceMessage)) {
-			acceptor.accept(createCompletionProposal((el as Actor).name, (el as Actor).name, null, context))
+			createAcceptorForParticipants(el, acceptor, context)
+		}
+	}
+
+	override completeSequenceMessage_Name(EObject model, Assignment assignment, ContentAssistContext context,
+		ICompletionProposalAcceptor acceptor) {
+		for (String el : messagesDefinedBefore(model as SequenceMessage)) {
+			acceptor.accept(createCompletionProposal("\"" + el + "\"", "\"" + el + "\"", null, context))
+
 		}
 	}
 
+	def createAcceptorForParticipants(EObject el, ICompletionProposalAcceptor acceptor, ContentAssistContext context) {
```

Comment:
Here can cast el as Participant and do only (el as Participant).name

Reply:
done